pkg/apis/batch/validation/validation.go

```diff
@@ -384,9 +386,43 @@ func validateTimeZone(timeZone *string, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(fldPath, timeZone, err.Error()))
 	}
 
+	if err := validateTimeZoneCase(*timeZone); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, timeZone, err.Error()))
+	}
+
 	return allErrs
 }
 
+// validateTimeZoneCase compares the timeZone with golang's builtin database
+// to ensure the caseing is correctly recognized. This only applies to MacOS
+// systems with case insensitive filesystem, where timezone can be
+// inproperly recognized.
+// See https://github.com/golang/go/issues/21512 for more details.
+func validateTimeZoneCase(timeZone string) error {
```

Comment:
would it make sense to have some unit tests for this function and passing the OS as a parameter?

Reply:
Seems like an overkill, but I could.

Reply:
ack, I will let it up to you

Reply:
Wait,  don't all operating systems have the potential to have case insensitive filesystems?

Reply:
Yeah, after some research it turns out that all can, but linux usually  defaults to case sensitive, whereas mac insensitive (https://en.wikipedia.org/wiki/Case_sensitivity#In_filesystems).